### ISSUE TITLE
Slopes for the finest mesh in a 2D flat plate case

### ISSUE THREAD

**Issue by JihooKang-KOR** [JihooKang-KOR/Data_driven_wall_modeling#1]
Dear @AndreWeiner,

With reference to the subject, I made a code to extract wall slopes and face slopes in a 2D flat plate case (commit 1bab3b1). Then, I extracted the related data from the case y+ = 0.05. Subsequently, I made two ML models for a correction test as follows.

1. Wall slope model
a. features : x-coordinates, Ux at 1st face
b. label : wall slope

2. Face slope model
a. features : x-coordinates, y-coordinates at faces, integral average Ux
b. label : face slope

At first glance, the slope values in the 2D case look similar to the values in the 1D channel case when the 2D case converged (at 3539th step). However, at the beginning of the simulation, the slopes at the wall are much larger, whereas the slopes at the faces are much smaller than the range of the 1D case. Therefore, when the 2D ML models were applied to the correction, an over-correction happened at the wall, an under-correction happened at the 1st face, compared to using the 1D mapping data.

When we use the 1D mapping data or the slope calculation of Spalding's function, the correction at the 1st face strongly exerts on the face, and thus the gap of the velocities between the 1st cell center and the 1st cell face is almost zero (then, the simulation crashes).

The situation for the 2D ML model is, however, opposite. In this case, the correction at the wall is too strong, so the simulation explodes at the wall.

I guess the slopes in the 1D mapping only fit for near the last time step, and hence the model cannot reflect the correct slopes at the beginning of the simulation. Therefore, I think the amount of the correction should gradually be decreased for the wall, and be increased for the 1st face. For instance, 2.0\*wallslope at the beginning, then changes gradually to 1.0\*wallslope after the certain steps, while 0.2\*faceslope at the beginning, then to 1.0\*faceslope. But it is only a rough and vague idea.

Please refer to the files and folders in the **"2DMapping"** folder in the cloud as follows.
1. The notebook **"2DFlatPlateMapping.ipynb"**
Contains two 2D mapping models in the notebook.
2. The folder **"turbulentFlatPlate_noWallFunc_ref0.05_datasets"**
Slope datasets with the finest mesh simulation in the folder. **"extractData_wall_2D.csv"** and **"extractData_face_2D.csv"** are the dataset csv files.
3. The folder **"turbulentFlatPlate_2Dmodel"**
The simulation by using the 2D mapping model.

Thank you very much for reading this issue.

Best regards,
Jihoo Kang

**Comment by AndreWeiner** [comment]
Hi Jihoo,
thanks for the update. Could you show some visualizations of your results?
Best, Andre

**Comment by JihooKang-KOR** [comment]
Dear @AndreWeiner,

Please see the figure as follows:
![image](https://user-images.githubusercontent.com/75588742/148625266-a9eaaff4-cc61-4e7a-b5d9-aa96321fd020.png)

The simulation was executed until 500th step for y+ = 50 and 100. As shown in the figure, the profile is already out of the range (exceeds 0.006). If the simulation kept going on, it would crash.

Therefore, I didn't try the full simulation for the 2D mapping model.

Best regards,
Jihoo Kang

**Comment by JihooKang-KOR** [comment]
Dear @AndreWeiner,

I realized that the 2D mapping model was wrongly set. Therefore, I changed the feature setting for wall as follows:

1. Wall slope model
a. features : x-coordinates, **y-coordinates at faces, integral average Ux**, ~~Ux at 1st face~~
b. label : wall slope
2. Face slope model
a. features : x-coordinates, y-coordinates at faces, integral average Ux
b. label : face slope

I trained these two models with 5000 epochs, and the notebook can be found in the same location of the cloud folder.

The result Cf is given by:
![image](https://user-images.githubusercontent.com/75588742/148709054-20cf7021-b010-433d-aaab-4f0fe511984d.png)

The simulation crashed, and the Cf values couldn't capture the proper behavior as above. It seems to be okay in the middle of the plate, but magGradUf is too small which leads to very high nuEff at the 1st face. If we see the csv files in the simulation folder (turbulentFlatPlate_2Dmodel/yplus_100 or yplus_50), the phenomenon is basically the same with the 1D mapping model. It means that the face slopes by the ML model becomes greater than the magGradUf after a certain time step (within 7~8th step).

Hence, I concluded that the slopes at the wall and the faces from the 1D mapping model have reasonable values compared to the 2D mapping model although they are not exactly the same.

In the cloud folder, I updated what I did, and thus the folder locations are the same as before.

I hope there would be another idea to alleviate this problem like Issue #2.

Best regards,
Jihoo Kang
